invoice/src/main/java/org/killbill/billing/invoice/generator/FixedAndRecurringInvoiceItemGenerator.java

```diff
@@ -430,11 +447,25 @@ private InvoiceItem generateFixedPriceItem(final UUID invoiceId, final UUID acco
                 Preconditions.checkNotNull(currentPlan, String.format("Unexpected null Plan name event = %s", thisEvent));
 
                 final DateTime catalogEffectiveDate = thisEvent.getCatalogEffectiveDate() != null ? thisEvent.getCatalogEffectiveDate() : null;
+                LocalDate invoiceItemEndDate = null;
+                if (thisEvent.getRecurringPrice() == null && thisEvent.getPlanPhase().getPhaseType() != PhaseType.EVERGREEN) {
+                    try {
+                        if (nextEvent != null && nextEvent.getSubscriptionId().equals(thisEvent.getSubscriptionId()) && (nextEvent.getTransitionType() == SubscriptionBaseTransitionType.PHASE)) {
```

Comment:
I would suggest here to not pass a `nextEvent` to this method when this is not the same subscription - similar to what we do for [processRecurringBillingEvents](https://github.com/killbill/killbill/blob/master/invoice/src/main/java/org/killbill/billing/invoice/generator/FixedAndRecurringInvoiceItemGenerator.java#L154).

Reply:
Fixed

Reply:
It seems like the condition `nextEvent.getSubscriptionId().equals(thisEvent.getSubscriptionId()` is not necessary is we guarantee that this method does not take events, i.e. `thisEvent` and `nextEvent` that cross subscriptions. Can you add the `@Nullable` to the argument and remove this extra condition?